airflow/providers/google/cloud/transfers/azure_fileshare_to_gcs.py

```diff
@@ -0,0 +1,217 @@
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import warnings
+from tempfile import NamedTemporaryFile
+from typing import Optional, Union, Sequence, Iterable
+
+from airflow import AirflowException
+from airflow.models import BaseOperator
+from airflow.providers.google.cloud.hooks.gcs import _parse_gcs_url, GCSHook
+from airflow.providers.microsoft.azure.hooks.azure_fileshare import AzureFileShareHook
+from airflow.utils.decorators import apply_defaults
+
+
+class AzureFileShareToGCSOperator(BaseOperator):
+    """
+    Synchronizes a Azure FileShare directory content (excluding subdirectories),
+    possibly filtered by a prefix, with a Google Cloud Storage destination path.
+
+    :param share_name: The Azure FileShare share where to find the objects. (templated)
+    :type share_name: str
+    :directory_name: (Optional) Path to Azure FileShare directory which content is to be transfered.
+        Defaults to root directory (templated)
+    :param prefix: Prefix string which filters objects whose name begin with
+        such prefix. (templated)
+    :type prefix: str
+    :param wasb_conn_id: The source WASB connection
+    :type wasb_conn_id: str
+    :param gcp_conn_id: (Optional) The connection ID used to connect to Google Cloud.
+    :type gcp_conn_id: str
+    :param dest_gcs_conn_id: (Deprecated) The connection ID used to connect to Google Cloud.
+        This parameter has been deprecated. You should pass the gcp_conn_id parameter instead.
+    :type dest_gcs_conn_id: str
+    :param dest_gcs: The destination Google Cloud Storage bucket and prefix
+        where you want to store the files. (templated)
+    :type dest_gcs: str
+    :param delegate_to: Google account to impersonate using domain-wide delegation of authority,
+        if any. For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    :param replace: Whether you want to replace existing destination files
+        or not.
+    :type replace: bool
+    :param gzip: Option to compress file for upload
+    :type gzip: bool
+    :param google_impersonation_chain: Optional Google service account to impersonate using
+        short-term credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    :type google_impersonation_chain: Union[str, Sequence[str]]
+
+
+    **Example**:
```

Comment:
The example below seems a little bit outdated - it uses wrong/deprecated attributes and does not list all templated_fields.

Reply:
fixd

Reply:
Hm, it seems you have not pushed your changes back to remote because I do not see any new changes.